pulp_2_tests/tests/rpm/api_v2/test_content_applicability.py

```diff
@@ -259,3 +283,206 @@ def test_negative(self):
                 applicability[0]['consumers'],
                 [consumer['consumer']['id']],
             )
+
+
+class ModularApplicabilityTestCase(unittest.TestCase):
+    """Perform modular repo applicability generation tasks.
+
+    Specifically, do the following:
+
+    1. Create a consumer.
+    2. Bind the consumer to the modular repository
+       ``RPM_WITH_MODULES_FEED_URL``.
+    3. Create a consumer profile with:
+       * List of RPMs
+       * List of Modules.
+    4. Regenerate applicability for the consumer.
+    5. Fetch applicability for the consumer. Verify that the packages
+       are eligible for upgrade.
+
+    This test targets the following:
+
+    * `Pulp #4158 <https://pulp.plan.io/issues/4158>`_
+    * `Pulp #4179 <https://pulp.plan.io/issues/4179>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create and sync a repository.
+
+        The regular test methods that run after this can create consumers that
+        bind to this repository.
+        """
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.18'):
+            raise unittest.SkipTest('This test requires Pulp 2.18 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_modular_rpm(self):
+        """Verify content is made available if appropriate.
+
+        This Tests the following:
+
+        1. Create a consumer profile with RPM less version than the
```

Comment:
```suggestion
        1. Create a consumer profile with an RPM version less than the
```
tl;dr - You choose. I gave you what I suggest above after asking other experts. Explanation below.  :) 
---
While `version` may be implied, it reads smoother to see the `RPM version` of the consumer profile first. Then, comparing to the module make more sense in the weird way English works. Else, it reads "a consumer profile with an RPM-less version" or an RPM without a version.

Changing `a RPM` to `an RPM` is a contentious change, but preferred by American English speakers. Typically, the indefinite article, `a` or `an`, if preceding a vowel sound is `an` or a consonant sound with `a`. However, my trusted native English speaking friends (who have BAs in Language Arts) tell me that `RPM` has a vowel sound... and "sounds better with `an`". In short, it depends if it `sounds` like a vowel or consonant, not if it IS a vowel or consonant. 

>  R is pronounced "aaaarr" with an open (vowel) sound. The same way you drive `an STi`. S is pronounced "ess" S and R are both consonants, but pronounced with open vowel sounds. You may get `a UTI`. U is a vowel, but pronounced with a Y consonant sound, therefore `a` instead of `an`.

I always hated the a/an rule, so I thought I would give me input above and bother a Bachelor's of Arts graduate.